pkg/npm/npm.go

```diff
@@ -206,24 +210,34 @@ func (exec *Execute) executeScript(packageJSON string, script string, runOptions
 }
 
 // FindPackageJSONFiles returns a list of all package.json fileUtils of the project excluding node_modules and gen/ directories
-func (exec *Execute) FindPackageJSONFiles() []string {
+func (exec *Execute) FindPackageJSONFiles(excludeList []string) ([]string, error) {
 	unfilteredListOfPackageJSONFiles, _ := exec.Utils.Glob("**/package.json")
 
+	nodeModulesExclude := filepath.Join("**", "node_modules", "**")
+	genExclude := filepath.Join("**", "gen", "**")
+	excludeList = append(excludeList, nodeModulesExclude, genExclude)
```

Comment:
I am wondering about `excludeList` which originates from user config. I think it can happen that `excludeList` may now be a mix of paths with forward and backward slashes. IIRC, `doublestar.Glob()` can take forward slashes on Windows, but the returned paths will contain backward slashes. Maybe use `doublestar.Match()` for processing the excludes, and that may already solve this problem?

Reply:
I have used the `doublestar.PathMatch()` on purpose here, quoting from the docs: Note: Match() is meant to be a drop-in replacement for path.Match(). As such, it always uses / as the path separator. If you are writing code that will run on systems where / is not the path separator (such as Windows), you want to use PathMatch() (below) instead.

Here it would not even be necessary to use filepath.join, since PathMatch() converts to `/` as seperator anyways.

Reply:
I'll change it to not use `filepath.join` to make it more readable.

Reply:
You could add a test-case for this: Fill the mock file-system with some package-json files, then pass an exclude list that has the wrong separator, but which should still exclude files in the mock FS. If it still works to exclude them, then you are sure.